experimental/packages/otlp-transformer/src/trace/index.ts

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import type { InstrumentationLibrary } from '@opentelemetry/core';
+import type { Resource } from '@opentelemetry/resources';
+import type { ReadableSpan } from '@opentelemetry/sdk-trace-base';
+import { toAttributes } from '../common/internal';
+import { toSpan } from './internal';
+import { IExportTraceServiceRequest } from './types';
+
+export function createExportTraceServiceRequest(spans: ReadableSpan[]): IExportTraceServiceRequest | null {
+  if (spans.length === 0) {
+    return null;
+  }
+
+  return {
+    resourceSpans: groupByResource(spans).map(({ resource, resourceSpans, schemaUrl: resourceSchemaUrl }) => ({
+      resource: {
+        attributes: toAttributes(resource.attributes),
+        droppedAttributesCount: 0,
+      },
+      instrumentationLibrarySpans: resourceSpans.map(({ instrumentationLibrary, instrumentationLibrarySpans, schemaUrl: librarySchemaUrl }) => ({
+        instrumentationLibrary: {
+          name: instrumentationLibrary.name,
+          version: instrumentationLibrary.version,
+        },
+        spans: instrumentationLibrarySpans.map(toSpan),
+        schemaUrl: librarySchemaUrl,
+      })),
+      schemaUrl: resourceSchemaUrl,
+    }))
+  };
+}
+
+type ResourceSpans = {
```

Comment:
```suggestion
interface ResourceSpans {
```

Reply:
What is the motivation for that?

Reply:
The differences between interface and type aliases may be trivial. See https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#differences-between-type-aliases-and-interfaces. An interface can only describe a shape of objects, whilst a type alias can describe anything (including primitives and literal types), I'd find it is more intuitive to use the limited interface syntax to describe an object shape.

This is just a style issue, as we don't perform any declaration merging. So I'm fine with both styles but 0.5 cents for the `interface` syntax.

Reply:
I usually use `type` when describing the structure of data for a parameter or return value, and use `interface` when describing something I expect to be implemented by a class and potentially have multiple class implementations.